src/main/kotlin/nl/info/client/brp/BrpClientService.kt

```diff
@@ -73,17 +73,21 @@ class BrpClientService @Inject constructor(
         private val FIELDS_PERSOON_BEPERKT = listOf(BURGERSERVICENUMMER, GESLACHT, NAAM, GEBOORTE, ADRESSERING)
     }
 
-    fun queryPersonen(personenQuery: PersonenQuery, auditEvent: String): PersonenQueryResponse =
-        updateQuery(personenQuery).let { updatedQuery ->
-            personenApi.personen(
-                personenQuery = updatedQuery,
-                purpose = resolvePurposeFromContext(
-                    auditEvent,
-                    queryPersonenDefaultPurpose.getOrNull()
-                ) { it.brpDoelbindingen?.zoekWaarde },
-                auditEvent = auditEvent
-            )
+    fun queryPersonen(personenQuery: PersonenQuery, isBsnQuery: Boolean, auditEvent: String): PersonenQueryResponse {
```

Comment:
Does it make sense to have 2 separate functions instead: `bsnQueryPersonen` and `queryPersonen` to make the code a bit simpler/cleaner in each of them and avoid the `if`s ?